docs/spring-server/spring-schema.md

```diff
@@ -0,0 +1,115 @@
+---
+id: spring-schema
+title: Writing Schemas with Spring
+---
+
+In order to expose your queries, mutations and/or subscriptions in the GraphQL schema you simply need to create beans that
+implement corresponding marker interface and they will be automatically picked up by `graphql-kotlin-spring-server`
+auto-configuration library.
+
+```kotlin
+@Component
+class MyAwesomeQuery : Query {
+  fun myAwesomeQuery(): Widget { ... }
+}
+
+@Component
+class MyAwesomeMutation : Mutation {
+  fun myAwesomeMutation(widget: Widget): Widget { ... }
+}
+
+@Component
+class MyAwesomeSubscription : Subscription {
+  fun myAwesomeSubscription(): Publisher<Widget> { ... }
+}
+
+data class Widget(val id: Int, val value: String)
+```
+
+will result in a Spring Boot reactive GraphQL web application with following schema.
+
+```graphql
+schema {
+  query: Query
+  mutation: Mutation
+  subscription: Subscription
+}
+
+type Query {
+  myAwesomeQuery: Widget!
+}
+
+type Mutation {
+  myAwesomeMutation(widget: Widget!): Widget!
+}
+
+type Subscription {
+  myAwesomeSubscription: Widget!
+}
+
+type Widget {
+  id: Int!
+  value: String!
+}
```

Comment:
```suggestion
}

input Widget {
  id: Int!
  value: String!
}
```

Reply:
Nice catch 👍 there would be `Widget` and `WidgetInput` will fix it (copied it from intro page).